src/openrct2/world/Location.hpp

```diff
@@ -244,32 +244,32 @@ struct CoordsXYZ : public CoordsXY
     {
     }
 
-    const CoordsXYZ operator+(const CoordsXYZ& rhs) const
+    constexpr const CoordsXYZ operator+(const CoordsXYZ& rhs) const
     {
         return { x + rhs.x, y + rhs.y, z + rhs.z };
     }
 
-    const CoordsXYZ operator-(const CoordsXYZ& rhs) const
+    constexpr const CoordsXYZ operator-(const CoordsXYZ& rhs) const
     {
         return { x - rhs.x, y - rhs.y, z - rhs.z };
     }
 
-    bool operator==(const CoordsXYZ& other) const
+    constexpr bool operator==(const CoordsXYZ& other) const
     {
         return x == other.x && y == other.y && z == other.z;
     }
 
-    CoordsXYZ ToTileStart() const
+    constexpr CoordsXYZ ToTileStart() const
     {
         return { floor2(x, COORDS_XY_STEP), floor2(y, COORDS_XY_STEP), z };
     }
 
-    CoordsXYZ ToTileCentre() const
+    constexpr CoordsXYZ ToTileCentre() const
     {
         return ToTileStart() + CoordsXYZ{ COORDS_XY_HALF_TILE, COORDS_XY_HALF_TILE, 0 };
     }
 
-    void setNull()
+    constexpr void setNull()
```

Comment:
Since this function doesn't modify anything other than itself and doesn't return anything, what is the use of `constexpr` here? Is it required so that other `constexpr` functions can call this?

Reply:
Yeah in a constexpr context you require them to be constexpr.